README.md

```diff
@@ -2,11 +2,68 @@
 
 Welcome to PuterBot: GUI Process Automation with Transformers! We are working with a dataset of user input events, screenshots, and window events. Our task is to generate the appropriate InputEvent(s) based on the previously recorded InputEvents and associated Screenshots, such that the task in the recording is accomplished, while accounting for differences in screen resolution, window size, application behavior, etc.
 
-## Problem Statement
+## Setup
+
+```
+git clone https://github.com/MLDSAI/puterbot.git
+cd puterbot
+python3.10 -m venv .venv
+source .venv/bin/activate
+pip install wheel
```

Comment:
Can wheel not be in requirements.txt?

Reply:
> Can wheel not be in requirements.txt?

For some reason this fails for me, not sure why, maybe a dependency resolution ordering issue.

> Also pypiwin32 missing from requirements

Thanks! I assume this is required to run `pip install -e .` on Windows, is that right? Can you please provide the version you're using so that I can add it?

Reply:
win32api was missing when I tried to run the tests. I have python 3.11.3, maybe that's why the tests failed, should I use 3.10?

Reply:
Tried with 3.10.11, still failed. pypiwin32 version is 223

`========================================================================= warnings summary ========================================================================= 
tests/puterbot/test_events.py::test_merge_consecutive_mouse_click_events
  C:\Workspace\puterbot\tests\puterbot\test_events.py:77: SAWarning: relationship 'InputEvent.recording' will copy column recording.timestamp to column input_event.recording_timestamp, which conflicts with relationship(s): 'Recording.input_events' (copies recording.timestamp to input_event.recording_timestamp). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="input_events"' to the 'InputEvent.recording' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)
    event = InputEvent(**event_dict)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================================== short test summary info ====================================================================== 
FAILED tests/puterbot/test_events.py::test_merge_consecutive_mouse_click_events - AssertionError: assert [{'mouse_butt...9999997}, ...] == [{'mouse_butt...9999997...FAILED tests/puterbot/test_events.py::test_merge_consecutive_mouse_move_events - AssertionError: assert [{'mouse_dx':... 'move', ...}] == [{'mouse_dx':... 'move',...FAILED tests/puterbot/test_events.py::test_merge_consecutive_mouse_scroll_events - AssertionError: assert [{'mouse_x': ...000000000001}] == [{'mouse_x': ...tamp':...FAILED tests/puterbot/test_events.py::test_remove_redundant_mouse_move_events - AssertionError: assert [{'children':...: 1.375}, ...] == [{'children':...9999996},...FAILED tests/puterbot/test_events.py::test_merge_consecutive_keyboard_events - AssertionError: {'values_changed': {"root[2]['timestamp']": {'new_value': 0.5499999...FAILED tests/puterbot/test_events.py::test_merge_consecutive_keyboard_events__grouped - AssertionError: {'values_changed': {"root[2]['timestamp']": {'new_value': ...============================================================== 6 failed, 1 passed, 1 warning in 1.87s ============================================================== `

Reply:
Thank you! I believe it's https://github.com/mhammond/pywin32

Reply:
Added in 91942a3

Reply:
Tried that, still getting the failures. Looks like they are down to timestamps for example:

```
E             'timestamp': 5.775},
E         -  {'children': [{'mouse_pressed': True, 'timestamp': 11.88},
E         +  {'children': [{'mouse_pressed': True, 'timestamp': 11.880000000000003},
E         ?                                                          +++++++++++++
E         -                {'mouse_pressed': False, 'timestamp': 12.155000000000001},
E         ?                                                                       ^
E         +                {'mouse_pressed': False, 'timestamp': 12.155000000000003},
E         ?                                                                       ^
E         -                {'mouse_pressed': True, 'timestamp': 12.21},
E         +                {'mouse_pressed': True, 'timestamp': 12.210000000000003},
E         ?                                                          +++++++++++++
E         -                {'mouse_pressed': False, 'timestamp': 12.485000000000001}],
E         ?                                                                       ^
E         +                {'mouse_pressed': False, 'timestamp': 12.485000000000003}],
E         ?                                                                       ^
E             'mouse_button_name': 'left',
E             'name': 'doubleclick',
E             'timestamp': 11.275},
E            {'mouse_button_name': 'right',
E             'mouse_pressed': True,
E             'name': 'click',
E             'timestamp': 16.775},
E            {'mouse_button_name': 'right',
E             'mouse_pressed': False,
E             'name': 'click',
E             'timestamp': 17.049999999999997},
E         -  {'children': [{'mouse_pressed': True, 'timestamp': 23.759999999999998},
E         ?                                                         ^^^^^^^^^^^^^^
E         +  {'children': [{'mouse_pressed': True, 'timestamp': 23.76},
E         ?                                                         ^
E         -                {'mouse_pressed': False, 'timestamp': 24.034999999999997}],
E         ?                                                           ^^^^^^^^^^^^^
E         +                {'mouse_pressed': False, 'timestamp': 24.035}],
E         ?                   
```

Reply:
Interesting. Looks like this may be a platform specific issue with https://github.com/MLDSAI/puterbot/blob/main/puterbot/utils.py#L337.

Can you please create a new Issue and include:
- windows version
- python version
- steps to reproduce

Thank you!! 🙏